pact/root/coin.pact

```diff
@@ -464,23 +508,24 @@
 
   (defun enforce-reserved:bool (account:string guard:guard)
     @doc "Enforce reserved account name protocols."
-    (let ((r (check-reserved account)))
-      (if (= "" r) true
-        (if (= "k" r)
-          (enforce
-            (= (format "{}" [guard])
-               (format "KeySet {keys: [{}],pred: keys-all}"
-                       [(drop 2 account)]))
-            "Single-key account protocol violation")
-          (enforce false
-            (format "Unrecognized reserved protocol: {}" [r]))))))
+    (if (validate-principal guard account)
+      true
+      (let ((r (check-reserved account)))
```

Comment:
Unless there is a compelling reason we need to remove `check-reserved` and this code and only enforce `validate-principal`